Scripts/Proxies/TwoHandedProxyBase.cs

```diff
@@ -159,23 +174,51 @@ public virtual IEnumerator Start()
             if (trackedObjectInput == null && m_PlayerInput)
                 trackedObjectInput = m_PlayerInput.GetActions<TrackedObject>();
 
-            var leftProxyHelper = m_LeftHand.GetComponent<ProxyHelper>();
-            var rightProxyHelper = m_RightHand.GetComponent<ProxyHelper>();
-            this.ConnectInterfaces(ObjectUtils.AddComponent<ProxyAnimator>(leftProxyHelper.gameObject), leftProxyHelper.rayOrigin);
-            this.ConnectInterfaces(ObjectUtils.AddComponent<ProxyAnimator>(rightProxyHelper.gameObject), rightProxyHelper.rayOrigin);
+            m_PreviousLeftHandPosition = trackedObjectInput.leftPosition.vector3;
+            m_PreviousRightHandPosition = trackedObjectInput.rightPosition.vector3;
         }
 
-        public virtual void OnDestroy() { }
-
         public virtual void Update()
         {
             if (active)
             {
-                m_LeftHand.localPosition = trackedObjectInput.leftPosition.vector3;
+                var leftLocalPosition = trackedObjectInput.leftPosition.vector3;
+                m_LeftHand.localPosition = leftLocalPosition;
                 m_LeftHand.localRotation = trackedObjectInput.leftRotation.quaternion;
 
-                m_RightHand.localPosition = trackedObjectInput.rightPosition.vector3;
+                var rightLocalPosition = trackedObjectInput.rightPosition.vector3;
+                m_RightHand.localPosition = rightLocalPosition;
                 m_RightHand.localRotation = trackedObjectInput.rightRotation.quaternion;
+
+                if (m_SemitransparentLockRequest == null)
+                {
+                    const float minShakeMovementThreshold = 0.001f;
+                    const int movementDetectionIncreaseScalar = 125;
+                    var movementDelta = Vector3.SqrMagnitude(leftLocalPosition - m_PreviousLeftHandPosition);
+                    movementDelta += Vector3.SqrMagnitude(rightLocalPosition - m_PreviousRightHandPosition);
+                    movementDelta *= Time.unscaledDeltaTime * movementDetectionIncreaseScalar;
+                    if (movementDelta > minShakeMovementThreshold)
+                    {
+                        m_ShakeFrequency += movementDelta;
+                        if (m_ShakeFrequency > 0.1f)
+                        {
+                            var shakeRequest = new ProxyFeedbackRequest
+                            {
+                                control = VRInputDevice.VRControl.LocalPosition,
+                                node = Node.None,
+                                tooltipText = null,
+                                suppressExisting = true,
+                                proxyShaken = true
+                            };
```

Comment:
Can we re-use this request? Construct once and add over and over again? Come to think of it, I think we can do this with most feedback. Or maybe we just want to formalize into a FeedbackPool?